Hive external table, Bucket identification error

hive外表bucket识别错误，eg: hive location如此配置： oss://ssports-bigdata-dw.oss-cn-beijing-internal.aliyuncs.com/hive/datawarehouse/dwd_xi_ubehav_applog 会报bucket格式识别错误；只有修改成oss://ssports-bigdata-dw/hive/datawarehouse/dwd_xi_ubehav_applog 去掉地区才不报错。感觉是路径中的“.”导致解析错误

hive中创建oss存储hive外部表，location中带地域信息：
```
CREATE EXTERNAL TABLE IF NOT EXISTS oss_lineorder (
 lo_orderkey bigint,
 lo_linenumber int,
 lo_custkey int,
 lo_partkey int,
 lo_suppkey int,
 lo_orderdate int,
 lo_orderpriority varchar(16),
 lo_shippriority int,
 lo_quantity int,
 lo_extendedprice int,
 lo_ordtotalprice int,
 lo_discount int,
 lo_revenue int,
 lo_supplycost int,
 lo_tax int,
 lo_commitdate int,
 lo_shipmode varchar(11)
) row format delimited fields terminated by '|'
 stored as textfile
location 'oss://dorisdb-data.oss-cn-zhangjiakou.aliyuncs.com/ssb_csv_1g/lineorder';
```
创建成功后，hive中查询该表，select count(*)报错：
```
hive> select count(*) from oss_lineorder;
Query ID = root_20220219160655_92f3b972-b5f3-449f-b3f2-c3984c6573a2
Total jobs = 1
Launching Job 1 out of 1
Number of reduce tasks determined at compile time: 1
In order to change the average load for a reducer (in bytes):
  set hive.exec.reducers.bytes.per.reducer=<number>
In order to limit the maximum number of reducers:
  set hive.exec.reducers.max=<number>
In order to set a constant number of reducers:
  set mapreduce.job.reduces=<number>
Starting Job = job_1644301670407_0064, Tracking URL = http://emr-header-1.cluster-49155:20888/proxy/application_1644301670407_0064/
Kill Command = /usr/lib/hadoop-current/bin/mapred job  -kill job_1644301670407_0064
Hadoop job information for Stage-1: number of mappers: 2; number of reducers: 1
2022-02-19 16:07:08,102 Stage-1 map = 0%,  reduce = 0%
2022-02-19 16:08:09,055 Stage-1 map = 0%,  reduce = 0%
2022-02-19 16:09:02,321 Stage-1 map = 100%,  reduce = 100%
Ended Job = job_1644301670407_0064 with errors
Error during job, obtaining debugging information...
Job Tracking URL: http://emr-header-1.cluster-49155:20888/proxy/application_1644301670407_0064/
Examining task ID: task_1644301670407_0064_m_000000 (and more) from job job_1644301670407_0064

Task with the most failures(4):
-----
Task ID:
  task_1644301670407_0064_m_000000

URL:
  http://emr-header-1.cluster-49155:8088/taskdetails.jsp?jobid=job_1644301670407_0064&tipid=task_1644301670407_0064_m_000000
-----
Diagnostic Messages for this Task:
Error: java.io.IOException: java.lang.reflect.InvocationTargetException
	at org.apache.hadoop.hive.io.HiveIOExceptionHandlerChain.handleRecordReaderCreationException(HiveIOExceptionHandlerChain.java:97)
	at org.apache.hadoop.hive.io.HiveIOExceptionHandlerUtil.handleRecordReaderCreationException(HiveIOExceptionHandlerUtil.java:57)
	at org.apache.hadoop.hive.shims.HadoopShimsSecure$CombineFileRecordReader.initNextRecordReader(HadoopShimsSecure.java:272)
	at org.apache.hadoop.hive.shims.HadoopShimsSecure$CombineFileRecordReader.<init>(HadoopShimsSecure.java:218)
	at org.apache.hadoop.hive.shims.HadoopShimsSecure$CombineFileInputFormatShim.getRecordReader(HadoopShimsSecure.java:346)
	at org.apache.hadoop.hive.ql.io.CombineHiveInputFormat.getRecordReader(CombineHiveInputFormat.java:721)
	at org.apache.hadoop.mapred.MapTask$TrackedRecordReader.<init>(MapTask.java:175)
	at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:444)
	at org.apache.hadoop.mapred.MapTask.run(MapTask.java:349)
	at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:174)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1732)
	at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:168)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.hadoop.hive.shims.HadoopShimsSecure$CombineFileRecordReader.initNextRecordReader(HadoopShimsSecure.java:258)
	... 11 more
Caused by: java.io.IOException: java.io.IOException: [ErrorCode] : 112 , [ErrorType]:  [HostId]: oss-cn-zhangjiakou.aliyuncs.com [ErrorMessage]: [E112]Fail to select channel, Host is down [R1][E112]Fail to select channel, Host is down [R2][E112]Fail to select channel, Host is down [R3][E112]Fail to select channel, Host is down
	at com.aliyun.emr.fs.internal.ossnative.OssFileletSystem.open(OssFileletSystem.java:330)
	at com.aliyun.emr.fs.internal.ossnative.OssNativeStore.open(OssNativeStore.java:522)
	at com.aliyun.emr.fs.oss.JindoOssFileSystem.openCore(JindoOssFileSystem.java:187)
	at com.aliyun.emr.fs.common.AbstractJindoShimsFileSystem.open(AbstractJindoShimsFileSystem.java:292)
	at org.apache.hadoop.fs.FileSystem.open(FileSystem.java:899)
	at org.apache.hadoop.mapred.LineRecordReader.<init>(LineRecordReader.java:109)
	at org.apache.hadoop.mapred.TextInputFormat.getRecordReader(TextInputFormat.java:67)
	at org.apache.hadoop.hive.ql.io.CombineHiveRecordReader.<init>(CombineHiveRecordReader.java:99)
	... 16 more
Caused by: java.io.IOException: [ErrorCode] : 112 , [ErrorType]:  [HostId]: oss-cn-zhangjiakou.aliyuncs.com [ErrorMessage]: [E112]Fail to select channel, Host is down [R1][E112]Fail to select channel, Host is down [R2][E112]Fail to select channel, Host is down [R3][E112]Fail to select channel, Host is down
	at com.alibaba.jboot.JbootFuture.get(JbootFuture.java:168)
	at com.alibaba.jboot.future.ByteBufferFuture.get(ByteBufferFuture.java:31)
	at com.aliyun.emr.fs.internal.ossnative.OssFileletSystem.open(OssFileletSystem.java:324)
	... 23 more


FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask
MapReduce Jobs Launched:
Stage-Stage-1: Map: 2  Reduce: 1   HDFS Read: 0 HDFS Write: 0 FAIL
Total MapReduce CPU Time Spent: 0 msec
```
select * limit *正常：
```
hive> select * from oss_lineorder limit 10;
OK
1	1	7381	155190	828	19960102	5-LOW	0	17	2116823	17366547	4	2032150	74711	2	19960212	TRUCK
1	2	7381	67310	163	19960102	5-LOW	0	36	4598316	17366547	9	4184467	76638	6	19960228	MAIL
1	3	7381	63700	71	19960102	5-LOW	0	8	1330960	17366547	10	1197864	99822	2	19960305	REG AIR
1	4	7381	2132	943	19960102	5-LOW	0	28	2895564	17366547	9	2634963	62047	6	19960330	AIR
1	5	7381	24027	1625	19960102	5-LOW	0	24	2282448	17366547	10	2054203	57061	4	19960314	FOB
1	6	7381	15635	1368	19960102	5-LOW	0	32	4962016	17366547	7	4614674	93037	2	19960207	MAIL
2	1	15601	106170	1066	19961201	1-URGENT	0	38	4469446	4692918	0	4469446	70570	5	19970114	RAIL
3	1	24664	4297	1959	19931014	5-LOW	0	45	5405805	19384625	6	5081456	72077	0	19940104	AIR
3	2	24664	19036	1667	19931014	5-LOW	0	49	4679647	19384625	10	4211682	57301	0	19931220	RAIL
3	3	24664	128449	1409	19931014	5-LOW	0	27	3989088	19384625	6	3749742	88646	7	19931122	SHIP
Time taken: 0.172 seconds, Fetched: 10 row(s)
```

使用此hive表在starrocks中创建hive外表报错，bucket name解析格式不对：
```
mysql> create external table ex_hive_oss_lineorder
    ->  (lo_orderkey bigint,
    ->  lo_linenumber int,
    ->  lo_custkey int,
    ->  lo_partkey int,
    ->  lo_suppkey int,
    ->  lo_orderdate int,
    ->  lo_orderpriority varchar(16),
    ->  lo_shippriority int,
    ->  lo_quantity int,
    ->  lo_extendedprice int,
    ->  lo_ordtotalprice int,
    ->  lo_discount int,
    ->  lo_revenue int,
    ->  lo_supplycost int,
    ->  lo_tax int,
    ->  lo_commitdate int,
    ->  lo_shipmode varchar(11))
    ->  engine=hive
    ->  properties (
    ->  "resource" = "hive_emr_tn",
    ->  "table" = "oss_lineorder",
    ->  "database" = "tn_test");
Query OK, 0 rows affected (0.05 sec)

mysql> select count(*) from ex_hive_oss_lineorder;
ERROR 1064 (HY000): com.starrocks.common.DdlException: get partition detail failed: com.starrocks.common.DdlException: get hive partition meta data failed: The bucket name "dorisdb-data.oss-cn-zhangjiakou.aliyuncs.com" is invalid. A bucket name must: 1) be comprised of lower-case characters, numbers or dash(-); 2) start with lower case or numbers; 3) be between 3-63 characters long.
```

报错对应FE日志：
```
2022-02-18 21:51:35,224 WARN (starrocks-mysql-nio-pool-44|789) [StatisticsCalculator.getTableRowCount():501] compute hive table row count failed : com.starrocks.common.DdlException: com.starrocks.common.DdlException: get partition detail failed: com.starrocks.common.DdlException: get hive partition meta data failed: The bucket name "dorisdb-data.oss-cn-zhangjiakou.aliyuncs.com" is invalid. A bucket name must: 1) be comprised of lower-case characters, numbers or dash(-); 2) start with lower case or numbers; 3) be between 3-63 characters long.
2022-02-18 21:51:35,224 WARN (starrocks-mysql-nio-pool-44|789) [StmtExecutor.execute():304] New planner error: select count(*) from ex_hive_oss_lineorder
com.starrocks.sql.common.StarRocksPlannerException: com.starrocks.common.DdlException: get partition detail failed: com.starrocks.common.DdlException: get hive partition meta data failed: The bucket name "dorisdb-data.oss-cn-zhangjiakou.aliyuncs.com" is invalid. A bucket name must: 1) be comprised of lower-case characters, numbers or dash(-); 2) start with lower case or numbers; 3) be between 3-63 characters long.
        at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.getTableRowCount(StatisticsCalculator.java:502) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.computeHiveScanNode(StatisticsCalculator.java:263) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.visitLogicalHiveScan(StatisticsCalculator.java:250) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.visitLogicalHiveScan(StatisticsCalculator.java:124) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.operator.logical.LogicalHiveScanOperator.accept(LogicalHiveScanOperator.java:123) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.estimatorStats(StatisticsCalculator.java:140) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.task.DeriveStatsTask.execute(DeriveStatsTask.java:76) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.task.SeriallyTaskScheduler.executeTasks(SeriallyTaskScheduler.java:40) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.memoOptimize(Optimizer.java:115) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.optimize(Optimizer.java:97) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.createQueryPlan(StatementPlanner.java:80) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:47) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:290) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:268) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:415) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:651) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_262]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_262]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_262]        
2022-02-18 21:51:35,224 WARN (starrocks-mysql-nio-pool-44|789) [StmtExecutor.execute():451] execute Exception, sql select count(*) from ex_hive_oss_lineorder
com.starrocks.sql.common.StarRocksPlannerException: com.starrocks.common.DdlException: get partition detail failed: com.starrocks.common.DdlException: get hive partition meta data failed: The bucket name "dorisdb-data.oss-cn-zhangjiakou.aliyuncs.com" is invalid. A bucket name must: 1) be comprised of lower-case characters, numbers or dash(-); 2) start with lower case or numbers; 3) be between 3-63 characters long.
        at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.getTableRowCount(StatisticsCalculator.java:502) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.computeHiveScanNode(StatisticsCalculator.java:263) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.visitLogicalHiveScan(StatisticsCalculator.java:250) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.visitLogicalHiveScan(StatisticsCalculator.java:124) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.operator.logical.LogicalHiveScanOperator.accept(LogicalHiveScanOperator.java:123) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.estimatorStats(StatisticsCalculator.java:140) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.task.DeriveStatsTask.execute(DeriveStatsTask.java:76) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.task.SeriallyTaskScheduler.executeTasks(SeriallyTaskScheduler.java:40) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.memoOptimize(Optimizer.java:115) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.optimize(Optimizer.java:97) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.createQueryPlan(StatementPlanner.java:80) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:47) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:290) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:268) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:415) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:651) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_262]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_262]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_262]    
```

--- a/be/src/common/config.h
+++ b/be/src/common/config.h
@@ -657,11 +657,11 @@ CONF_Bool(use_hdfs_pread, "true");
 // default: true
 CONF_Bool(rewrite_partial_segment, "true");
 
-// Properties to access aws s3.
-CONF_String(aws_access_key_id, "");
-CONF_String(aws_secret_access_key, "");
-CONF_String(aws_s3_endpoint, "");
-CONF_Int64(aws_s3_max_connection, "102400");
+// Properties to access object storage
+CONF_String(object_storage_access_key_id, "");
+CONF_String(object_storage_secret_access_key, "");
+CONF_String(object_storage_endpoint, "");
+CONF_Int64(object_storage_max_connection, "102400");
 
 CONF_Bool(enable_orc_late_materialization, "true");
 

--- a/be/src/env/env_s3.cpp
+++ b/be/src/env/env_s3.cpp
@@ -78,17 +78,14 @@ StatusOr<std::unique_ptr<RandomAccessFile>> EnvS3::new_random_access_file(const 
     Aws::Client::ClientConfiguration config;
     config.scheme = Aws::Http::Scheme::HTTP;
     if (is_oss_path(namenode.c_str())) {
-        assert(!config::object_storage_endpoint.empty());
+        CHECK(!config::object_storage_access_key_id.empty()) << "Configuration object_storage_access_key_id is missing for OSS.";
+        CHECK(!config::object_storage_secret_access_key.empty()) << "Configuration object_storage_secret_access_key is missing for OSS.";
+        CHECK(!config::object_storage_endpoint.empty()) << "Configuration object_storage_endpoint is missing for OSS.";
         config.endpointOverride = get_endpoint_from_oss_bucket(config::object_storage_endpoint, &bucket);
-    }
-    if (is_s3a_path(namenode.c_str()) && !config::object_storage_endpoint.empty()){
+    } else if (!config::object_storage_endpoint.empty()){
         config.endpointOverride = config::object_storage_endpoint;
     }
     config.maxConnections = config::object_storage_max_connection;
-    if (is_oss_path(namenode.c_str())) {
-        assert(!config::object_storage_access_key_id.empty());
-        assert(!config::object_storage_secret_access_key.empty());
-    }
     S3ObjectStore store(config);
     RETURN_IF_ERROR(store.init(false));
     ASSIGN_OR_RETURN(auto input_file, store.get_object(bucket, object));

--- a/be/src/env/env_s3.cpp
+++ b/be/src/env/env_s3.cpp
@@ -78,11 +78,13 @@ StatusOr<std::unique_ptr<RandomAccessFile>> EnvS3::new_random_access_file(const 
     Aws::Client::ClientConfiguration config;
     config.scheme = Aws::Http::Scheme::HTTP;
     if (is_oss_path(namenode.c_str())) {
-        CHECK(!config::object_storage_access_key_id.empty()) << "Configuration object_storage_access_key_id is missing for OSS.";
-        CHECK(!config::object_storage_secret_access_key.empty()) << "Configuration object_storage_secret_access_key is missing for OSS.";
+        CHECK(!config::object_storage_access_key_id.empty())
+                << "Configuration object_storage_access_key_id is missing for OSS.";
+        CHECK(!config::object_storage_secret_access_key.empty())
+                << "Configuration object_storage_secret_access_key is missing for OSS.";
         CHECK(!config::object_storage_endpoint.empty()) << "Configuration object_storage_endpoint is missing for OSS.";
         config.endpointOverride = get_endpoint_from_oss_bucket(config::object_storage_endpoint, &bucket);
-    } else if (!config::object_storage_endpoint.empty()){
+    } else if (!config::object_storage_endpoint.empty()) {
         config.endpointOverride = config::object_storage_endpoint;
     }
     config.maxConnections = config::object_storage_max_connection;

--- a/be/src/env/env_s3.cpp
+++ b/be/src/env/env_s3.cpp
@@ -77,15 +77,20 @@ StatusOr<std::unique_ptr<RandomAccessFile>> EnvS3::new_random_access_file(const 
     std::string object = path.substr(namenode.size(), path.size() - namenode.size());
     Aws::Client::ClientConfiguration config;
     config.scheme = Aws::Http::Scheme::HTTP;
-    if (!config::aws_s3_endpoint.empty()) {
-        config.endpointOverride = config::aws_s3_endpoint;
+    if (is_oss_path(namenode.c_str())) {
+        assert(!config::object_storage_endpoint.empty());
+        config.endpointOverride = get_endpoint_from_oss_bucket(config::object_storage_endpoint, &bucket);
     }
-    config.maxConnections = config::aws_s3_max_connection;
-    S3Credential cred;
-    cred.access_key_id = config::aws_access_key_id;
-    cred.secret_access_key = config::aws_secret_access_key;
+    if (is_s3a_path(namenode.c_str()) && !config::object_storage_endpoint.empty()){
+        config.endpointOverride = config::object_storage_endpoint;
+    }
+    config.maxConnections = config::object_storage_max_connection;
+    if (is_oss_path(namenode.c_str())) {
+        assert(!config::object_storage_access_key_id.empty());
+        assert(!config::object_storage_secret_access_key.empty());
+    }
     S3ObjectStore store(config);
-    RETURN_IF_ERROR(store.init(&cred, false));
+    RETURN_IF_ERROR(store.init(false));
     ASSIGN_OR_RETURN(auto input_file, store.get_object(bucket, object));
     return std::make_unique<RandomAccessFileWrapper>(std::move(input_file), path);
 }

--- a/be/src/env/env_s3.cpp
+++ b/be/src/env/env_s3.cpp
@@ -78,11 +78,12 @@ StatusOr<std::unique_ptr<RandomAccessFile>> EnvS3::new_random_access_file(const 
     Aws::Client::ClientConfiguration config;
     config.scheme = Aws::Http::Scheme::HTTP;
     if (is_oss_path(namenode.c_str())) {
-        CHECK(!config::object_storage_access_key_id.empty())
-                << "Configuration object_storage_access_key_id is missing for OSS.";
-        CHECK(!config::object_storage_secret_access_key.empty())
-                << "Configuration object_storage_secret_access_key is missing for OSS.";
-        CHECK(!config::object_storage_endpoint.empty()) << "Configuration object_storage_endpoint is missing for OSS.";
+        if (config::object_storage_access_key_id.empty() ||
+            config::object_storage_secret_access_key.empty() ||
+            config::object_storage_endpoint.empty()) {
+            return Status::RuntimeError("Configuration object_storage_access_key_id, "
+                    "object_storage_secret_access_key and object_storage_endpoint is required for OSS.");
+        }
         config.endpointOverride = get_endpoint_from_oss_bucket(config::object_storage_endpoint, &bucket);
     } else if (!config::object_storage_endpoint.empty()) {
         config.endpointOverride = config::object_storage_endpoint;

--- a/be/src/env/env_s3.cpp
+++ b/be/src/env/env_s3.cpp
@@ -78,11 +78,11 @@ StatusOr<std::unique_ptr<RandomAccessFile>> EnvS3::new_random_access_file(const 
     Aws::Client::ClientConfiguration config;
     config.scheme = Aws::Http::Scheme::HTTP;
     if (is_oss_path(namenode.c_str())) {
-        if (config::object_storage_access_key_id.empty() ||
-            config::object_storage_secret_access_key.empty() ||
+        if (config::object_storage_access_key_id.empty() || config::object_storage_secret_access_key.empty() ||
             config::object_storage_endpoint.empty()) {
-            return Status::RuntimeError("Configuration object_storage_access_key_id, "
-                    "object_storage_secret_access_key and object_storage_endpoint is required for OSS.");
+            return Status::RuntimeError(
+                    "Configuration object_storage_access_key_id, object_storage_secret_access_key "
+                    "and object_storage_endpoint is required for OSS.");
         }
         config.endpointOverride = get_endpoint_from_oss_bucket(config::object_storage_endpoint, &bucket);
     } else if (!config::object_storage_endpoint.empty()) {

--- a/be/src/object_store/s3_object_store.cpp
+++ b/be/src/object_store/s3_object_store.cpp
@@ -35,17 +35,11 @@ static inline Aws::String to_aws_string(const std::string& s) {
 S3ObjectStore::S3ObjectStore(const Aws::Client::ClientConfiguration& config) : _config(config) {}
 
 Status S3ObjectStore::init(bool use_transfer_manager) {
-    const char* env_access_key_id = getenv("AWS_ACCESS_KEY_ID");
-    const char* env_secret_access_key = getenv("AWS_SECRET_ACCESS_KEY");
-    const string be_conf_access_key_id = config::object_storage_access_key_id;
-    const string be_conf_secret_access_key = config::object_storage_secret_access_key;
-    if (!be_conf_access_key_id.empty() && !be_conf_secret_access_key.empty()) {
+    const string access_key_id = config::object_storage_access_key_id;
+    const string secret_access_key = config::object_storage_secret_access_key;
+    if (!access_key_id.empty() && !secret_access_key.empty()) {
         std::shared_ptr<Aws::Auth::AWSCredentialsProvider> credentials =
-                std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(be_conf_access_key_id, be_conf_secret_access_key);
-        _client = std::make_shared<Aws::S3::S3Client>(credentials, _config);
-    } else if (strlen(env_access_key_id) > 0 && strlen(env_secret_access_key) > 0) {
-        std::shared_ptr<Aws::Auth::AWSCredentialsProvider> credentials =
-                std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(env_secret_access_key, env_secret_access_key);
+                std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(access_key_id, secret_access_key);
         _client = std::make_shared<Aws::S3::S3Client>(credentials, _config);
     } else {
         // if not cred provided, we can use default cred in aws profile.

--- a/be/src/object_store/s3_object_store.cpp
+++ b/be/src/object_store/s3_object_store.cpp
@@ -34,11 +34,18 @@ static inline Aws::String to_aws_string(const std::string& s) {
 
 S3ObjectStore::S3ObjectStore(const Aws::Client::ClientConfiguration& config) : _config(config) {}
 
-Status S3ObjectStore::init(const S3Credential* cred, bool use_transfer_manager) {
-    if (cred != nullptr) {
+Status S3ObjectStore::init(bool use_transfer_manager) {
+    const char* env_access_key_id = getenv("AWS_ACCESS_KEY_ID");
+    const char* env_secret_access_key = getenv("AWS_SECRET_ACCESS_KEY");
+    const string be_conf_access_key_id = config::object_storage_access_key_id;
+    const string be_conf_secret_access_key = config::object_storage_secret_access_key;
+    if (!be_conf_access_key_id.empty() && !be_conf_secret_access_key.empty()) {
         std::shared_ptr<Aws::Auth::AWSCredentialsProvider> credentials =
-                std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(cred->access_key_id.c_str(),
-                                                                          cred->secret_access_key.c_str());
+                std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(be_conf_access_key_id, be_conf_secret_access_key);
+        _client = std::make_shared<Aws::S3::S3Client>(credentials, _config);
+    } else if (strlen(env_access_key_id) > 0 && strlen(env_secret_access_key) > 0) {
+        std::shared_ptr<Aws::Auth::AWSCredentialsProvider> credentials =
+                std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(env_secret_access_key, env_secret_access_key);
         _client = std::make_shared<Aws::S3::S3Client>(credentials, _config);
     } else {
         // if not cred provided, we can use default cred in aws profile.

--- a/be/src/object_store/s3_object_store.h
+++ b/be/src/object_store/s3_object_store.h
@@ -23,7 +23,7 @@ class S3ObjectStore final : public ObjectStore {
 public:
     S3ObjectStore(const Aws::Client::ClientConfiguration& config);
     ~S3ObjectStore() = default;
-    Status init(const S3Credential* cred = nullptr, bool use_transfer_manager = false);
+    Status init(bool use_transfer_manager = false);
 
     Status create_bucket(const std::string& bucket) override;
 

--- a/be/src/runtime/hdfs/hdfs_fs_cache.cpp
+++ b/be/src/runtime/hdfs/hdfs_fs_cache.cpp
@@ -3,13 +3,8 @@
 #include "runtime/hdfs/hdfs_fs_cache.h"
 
 #include <memory>
-#include <utility>
 
-#include "common/config.h"
 #include "gutil/strings/substitute.h"
-#ifdef STARROCKS_WITH_AWS
-#include "object_store/s3_object_store.h"
-#endif
 #include "util/hdfs_util.h"
 
 namespace starrocks {
@@ -26,25 +21,6 @@ static Status create_hdfs_fs_handle(const std::string& namenode, HdfsFsHandle* h
                                                              namenode, get_hdfs_err_msg()));
         }
 
-    } else if (is_s3a_path(nn) || is_oss_path(nn)) {
-#ifdef STARROCKS_WITH_AWS
-        handle->type = HdfsFsHandle::Type::S3;
-        Aws::Client::ClientConfiguration config;
-        config.scheme = Aws::Http::Scheme::HTTP;
-        if (!config::aws_s3_endpoint.empty()) {
-            config.endpointOverride = config::aws_s3_endpoint;
-        }
-        config.maxConnections = config::aws_s3_max_connection;
-
-        S3Credential cred;
-        cred.access_key_id = config::aws_access_key_id;
-        cred.secret_access_key = config::aws_secret_access_key;
-        auto store = std::make_unique<S3ObjectStore>(config);
-        RETURN_IF_ERROR(store->init(&cred, false));
-        handle->object_store = store.release();
-#else
-        return Status::NotSupported("Does not support S3");
-#endif
     } else {
         return Status::InternalError(strings::Substitute("failed to make client, namenode=$0", namenode));
     }

--- a/be/src/util/hdfs_util.cpp
+++ b/be/src/util/hdfs_util.cpp
@@ -64,6 +64,18 @@ std::string get_bucket_from_namenode(const std::string& namenode) {
     return namenode.substr(n, n2 - n);
 }
 
+std::string get_endpoint_from_oss_bucket(const std::string& default_bucket, std::string* bucket) {
+    auto endpoint_start_index = bucket->find('.');
+    if (endpoint_start_index == std::string::npos) {
+        return default_bucket;
+    }
+    endpoint_start_index = endpoint_start_index + 1;
+    auto endpoint_end_index = bucket->size();
+    std::string endpoint = bucket->substr(endpoint_start_index, endpoint_end_index - endpoint_start_index + 1);
+    *bucket = bucket->substr(0, endpoint_start_index - 1);
+    return endpoint;
+}
+
 static bool is_specific_path(const char* path, const char* specific_prefix) {
     size_t prefix_len = strlen(specific_prefix);
     return strncmp(path, specific_prefix, prefix_len) == 0;

--- a/be/src/util/hdfs_util.h
+++ b/be/src/util/hdfs_util.h
@@ -12,6 +12,7 @@ std::string get_hdfs_err_msg();
 
 Status get_namenode_from_path(const std::string& path, std::string* namenode);
 std::string get_bucket_from_namenode(const std::string& namenode);
+std::string get_endpoint_from_oss_bucket(const std::string& default_bucket, std::string* bucket);
 
 // Returns true if the path refers to a location on an HDFS filesystem.
 bool is_hdfs_path(const char* path);

--- a/fe/fe-core/src/main/java/com/starrocks/external/ObjectStorageUtils.java
+++ b/fe/fe-core/src/main/java/com/starrocks/external/ObjectStorageUtils.java
@@ -6,7 +6,6 @@ public class ObjectStorageUtils {
     private static final String SCHEME_S3A = "s3a://";
     private static final String SCHEME_S3 = "s3://";
     private static final String SCHEME_S3N = "s3n://";
-    private static final String SCHEME_OSS = "oss://";
     private static final String SCHEME_S3_PREFIX = "s3";
     private static final String SCHEME_OSS_PREFIX = "oss";
 
@@ -20,10 +19,6 @@ public class ObjectStorageUtils {
         }
         if (path.startsWith(SCHEME_S3N)) {
             return SCHEME_S3A + path.substring(SCHEME_S3N.length());
-        }
-        // use s3a to access oss.
-        if (path.startsWith(SCHEME_OSS)) {
-            return SCHEME_S3A + path.substring(SCHEME_OSS.length());
         }
         return path;
     }

